domain/model/watcher_test.go

```diff
@@ -0,0 +1,175 @@
+
+	modelName := "test-model"
+	var modelUUID model.UUID
+	var modelUUIDStr string
+
+	harness := watchertest.NewHarness(s, watchertest.NewWatcherC(c, watcher))
+
+	harness.AddTest(func(c *gc.C) {
+		// Create a new model named test-model and activate it.
+		modelUUID = modeltesting.CreateTestModel(c, s.TxnRunnerFactory(), modelName)
+		modelUUIDStr = modelUUID.String()
+
+		// Set model activated status to false. This should not trigger a change event as activation status is false.
+		s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+			res, err := tx.ExecContext(ctx, "UPDATE model SET activated = ? WHERE uuid = ?", false, modelUUID)
+			c.Assert(err, jc.ErrorIsNil)
+			rowsAffected, err := res.RowsAffected()
+			c.Assert(err, jc.ErrorIsNil)
+			c.Check(int(rowsAffected), gc.Equals, 1)
+			return nil
+		})
+
+		// Ensure that the initial activated status of model is indeed false.
+		var testModel dbModel
+		row := s.DB().QueryRow(`SELECT activated FROM model WHERE uuid = ?`, modelUUIDStr)
+		err = row.Scan(&testModel.Activated)
+		c.Assert(err, jc.ErrorIsNil)
+		c.Check(testModel.Activated, jc.IsFalse)
+
+	}, func(w watchertest.WatcherC[[]string]) {
+		// Get the change.
+		w.AssertNoChange()
+	})
+
+	harness.AddTest(func(c *gc.C) {
+		// Update model active status.
+		s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+			// Update activated status of model. This should trigger a change event.
+			res, err := tx.ExecContext(ctx, "UPDATE model SET activated = ? WHERE uuid = ?", true, modelUUIDStr)
+			c.Assert(err, jc.ErrorIsNil)
+			rowsAffected, err := res.RowsAffected()
+			c.Assert(err, jc.ErrorIsNil)
+			c.Check(int(rowsAffected), gc.Equals, 1)
+			return nil
+		})
+
+	}, func(w watchertest.WatcherC[[]string]) {
+		w.Check(
+			watchertest.StringSliceAssert(
+				modelUUIDStr,
+			),
+		)
+	})
+
+	harness.AddTest(func(c *gc.C) {
+		// Update model active status.
+		s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+			// Update name of model. This should trigger a change event.
+			res, err := tx.ExecContext(ctx, "UPDATE model SET name = ? WHERE uuid = ?", "new-test-model", modelUUIDStr)
+			c.Assert(err, jc.ErrorIsNil)
+			rowsAffected, err := res.RowsAffected()
+			c.Assert(err, jc.ErrorIsNil)
+			c.Check(int(rowsAffected), gc.Equals, 1)
+
+			// Checks if model activated status is updated to true successfully.
+			selectModelQuery := `
+			SELECT uuid, activated, model_type_id, name, cloud_uuid, life_id, owner_uuid 
+			FROM model 
+			WHERE uuid = ?`
+			var testModel dbModel
+			row := tx.QueryRow(selectModelQuery, modelUUIDStr)
+			err = row.Scan(&testModel.UUID, &testModel.Activated, &testModel.ModelTypeID,
+				&testModel.Name, &testModel.CloudUUID, &testModel.LifeID, &testModel.OwnerUUID)
+			c.Assert(err, jc.ErrorIsNil)
+			c.Check(testModel.UUID, gc.Equals, modelUUIDStr)
+			c.Check(testModel.Activated, jc.IsTrue)
```

Comment:
We are more interested in the new name here instead.

Reply:
We also should add a test: updating a not-activated model won't trigger an event.

Reply:
Good point, thanks for pointing these out Kelvin, will refactor accordingly :)